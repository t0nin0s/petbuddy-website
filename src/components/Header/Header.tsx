import React from "react";
import styled from 'styled-components'
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import Typography from "@material-ui/core/Typography";
import IconButton from "@material-ui/core/IconButton";
import MenuIcon from "@material-ui/icons/Menu";
import MenuItem from "@material-ui/core/MenuItem";
import Menu from "@material-ui/core/Menu";

const linksArray: Array<string> = ['Home', 'Cat Services', 'Dog Services', 'Small Pet Services', 'Become a PBuddy', 'Contact Us']

const StyledAppBar = styled(AppBar)`
`
const StyledToolbar = styled(Toolbar)`
background-color: whitesmoke;
color: purple;
justify-content:space-between
`
const StyledTypography = styled(Typography)`
&& {
  font-size:2rem;
  font-weight:700
}
`
const StyledIconButton = styled(IconButton)`
 `
const StyledMenuIcon = styled(MenuIcon)`
  && {
  font-size:3rem;
  font-weight:700;
  color: purple;
  }
`
const StyledMenu = styled(Menu)`
margin-left: 50%;
margin-top: 17%;
width:150%;

`

export default function MenuAppBar() {
  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
  const open = Boolean(anchorEl);

  const handleMenu = (event: React.MouseEvent<HTMLElement>) => {
    setAnchorEl(event.currentTarget);
  };

  const handleClose = () => {
    setAnchorEl(null);
  };

  const ourServices = linksArray.map(service => {
    return <MenuItem onClick={handleClose}>{service}</MenuItem>
  })

  return (
    <div className="header-container">
      <StyledAppBar position="static">
        <StyledToolbar>
          <StyledTypography >
            PetBuddy
          </StyledTypography>
          <StyledIconButton
            onClick={handleMenu}
          >
            <StyledMenuIcon />
          </StyledIconButton>
          <StyledMenu
            open={open}
            onClose={handleClose}
          >
            {ourServices}
          </StyledMenu>
        </StyledToolbar>
      </StyledAppBar>
    </div>
  );
}