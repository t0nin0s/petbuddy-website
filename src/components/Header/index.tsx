import React from "react";
import styled from 'styled-components'
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import Typography from "@material-ui/core/Typography";
import IconButton from "@material-ui/core/IconButton";
import MenuIcon from "@material-ui/icons/Menu";
import MenuItem from "@material-ui/core/MenuItem";
import Menu from "@material-ui/core/Menu";
import { mediaQueries } from '../../mediaqueries'

const linksArray: Array<string> = ['Cat Services', 'Dog Services', 'Small Pet Services', 'Become a PBuddy', 'Contact Us']

const StyledAppBar = styled(AppBar)``

const StyledToolbar = styled(Toolbar)`
  background-color: whitesmoke;
  color: purple;
  justify-content:space-between
`
const StyledTypography = styled(Typography)`
  font-size:2rem;
  font-weight:700
  `
const StyledIconButton = styled(IconButton)`
${mediaQueries("medium")`
  display:none;`}`

const StyledMenuIcon = styled(MenuIcon)`
  font-size:3rem;
  font-weight:700;
  color: purple;
  display:flex;
${mediaQueries("medium")`
  display:none;`}
`
const StyledMenu = styled(Menu)`
  margin-left: 3.5%;
`
const NavBarItems = styled.div`
display:none;
  ${mediaQueries("medium")`
    display:flex;
    flex-direction:row;
    flex-grow:1;
    justify-content: flex-end;
    color:#f48fb1;
    font-weight: 900;
    `}
`

export default function MenuAppBar() {
  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
  const open = Boolean(anchorEl);

  const handleMenu = (event: React.MouseEvent<HTMLElement>) => {
    setAnchorEl(event.currentTarget);
  };

  const handleClose = () => {
    setAnchorEl(null);
  };

  const ourServices = linksArray.map(service => {
    return <MenuItem onClick={handleClose}>{service}</MenuItem>
  })

  return (
    <div className="header-container">
      <StyledAppBar position="static">
        <StyledToolbar>
          <StyledTypography >
            PetBuddy
          </StyledTypography>
          <NavBarItems>{ourServices}</NavBarItems>
          <StyledIconButton
            onClick={handleMenu}>
            <StyledMenuIcon />
          </StyledIconButton>
          <StyledMenu
            anchorEl={anchorEl}
            getContentAnchorEl={null}
            anchorOrigin={{ vertical: "bottom", horizontal: "center" }}
            transformOrigin={{ vertical: "top", horizontal: "center" }}
            open={open}
            onClose={handleClose}
          >
            {ourServices}
          </StyledMenu>
        </StyledToolbar>
      </StyledAppBar>
    </div>
  );
}