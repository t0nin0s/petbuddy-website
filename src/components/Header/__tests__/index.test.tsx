import React from 'react'
import Header from '../index'
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('Header component', () => {
  it('should render', () => {
    const wrapper = shallow(<Header />)
    expect(toJson(wrapper)).toMatchSnapshot();
  })
  
})

