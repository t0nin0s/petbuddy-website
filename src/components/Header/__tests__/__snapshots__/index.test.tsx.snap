// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render 1`] = `
<Styled(WithStyles(ForwardRef(AppBar)))>
  <Styled(WithStyles(ForwardRef(Toolbar)))>
    <styled.div>
      <PetBuddyLogo
        size="small"
      />
    </styled.div>
    <Styled(WithStyles(ForwardRef(Typography)))>
      PetBuddy
    </Styled(WithStyles(ForwardRef(Typography)))>
    <_default />
    <Styled(WithStyles(ForwardRef(IconButton)))
      onClick={[Function]}
    >
      <Styled(Component) />
    </Styled(WithStyles(ForwardRef(IconButton)))>
    <Styled(WithStyles(ForwardRef(Menu)))
      PaperProps={
        Object {
          "style": Object {
            "textAlign": "center",
            "width": "100vw",
          },
        }
      }
      anchorEl={null}
      anchorOrigin={
        Object {
          "horizontal": "center",
          "vertical": "bottom",
        }
      }
      getContentAnchorEl={null}
      onClose={[Function]}
      open={false}
      transformOrigin={
        Object {
          "horizontal": "center",
          "vertical": "top",
        }
      }
    >
      <_default
        primary={true}
      />
    </Styled(WithStyles(ForwardRef(Menu)))>
  </Styled(WithStyles(ForwardRef(Toolbar)))>
</Styled(WithStyles(ForwardRef(AppBar)))>
`;
